src/common/entryPoint.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {TelemetryHelper} from "../common/telemetryHelper";
+import {Telemetry} from "../common/telemetry";
+import {Log} from "../common/Log";
+import {OutputChannel} from "vscode";
+
+/* This class should we used for each entry point of the code, so we handle telemetry and error reporting properly */
+export class EntryPoint {
```

Comment:
It's only intended to be used on entry points (where our app starts, and where other apps like VSCode is calling us). It's not a general class for telemetry and error reporting.